presto-main/src/main/java/com/facebook/presto/type/DecimalOperators.java

```diff
@@ -451,4 +452,22 @@ public static Slice negate(@SqlType("decimal(p, s)") Slice arg)
             return Decimals.encodeUnscaledValue(argBigInteger.negate());
         }
     }
+
+    @ScalarOperator(HASH_CODE)
+    public static final class HashCode
+    {
+        @LiteralParameters({"p", "s"})
+        @SqlType("bigint")
+        public static long hashCode(@SqlType("decimal(p, s)") long arg)
+        {
+            return Long.hashCode(arg);
```

Comment:
These will need to be updated to return 64-bit hashes as part of https://github.com/prestodb/presto/pull/4960